tests/e2e/volumeproperty_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is an end-to-end test. Test creates volumes of different format-types
+// and verifies their properties from ESX as well as docker host.
+// Properties being verified - capacity, disk-format and vm-attached field.
+
+// Test assumes that SSH cert has been setup to enable password-less login to VM and ESX.
+
+package e2e
+
+import (
+	"log"
+	"reflect"
+	"strings"
+
+	dockerconst "github.com/vmware/docker-volume-vsphere/tests/constants/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/govc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	testName           = "volumeproperty"
+	diskStatusDetached = "detached"
+	size               = "100MB"
+	diskFormatOption   = " -o diskformat="
+)
+
+type VolumePropertyTestSuite struct {
+	volumeNames           [3]string
+	dockerHostIP          string
+	dockerHostName        string
+	containerName         string
+	esxIP                 string
+	formatTypes           []string
+	dockerVolumeRemoveCmd string
+	volumeAttached        bool
+}
+
+func (s *VolumePropertyTestSuite) SetUpTest(c *C) {
+	s.dockerHostIP = inputparams.GetDockerHostIP("VM2")
+	s.dockerHostName = govc.RetrieveVMNameFromIP(s.dockerHostIP)
+	s.esxIP = inputparams.GetEsxIP()
+	s.containerName = inputparams.GetContainerNameWithTimeStamp(testName)
+	s.formatTypes = []string{"thin", "zeroedthick", "eagerzeroedthick"}
+	s.dockerVolumeRemoveCmd = dockerconst.RemoveVolume
+	s.verifyParams(c)
+}
+
+// verifyParams - skips the test if dockerHostIP/dockerHostName/esxIP are empty
+func (s *VolumePropertyTestSuite) verifyParams(c *C) {
+	if s.dockerHostIP == "" || s.dockerHostName == "" || s.esxIP == "" {
+		c.Skip("Empty values - hence skipping the test")
+	}
+}
+
+// TearDownTest - Removes all the containers and deletes the volumes
+func (s *VolumePropertyTestSuite) TearDownTest(c *C) {
+	out, err := dockercli.RemoveAllContainers(s.dockerHostIP)
+	c.Assert(err, IsNil, Commentf(out))
+	out, err = ssh.InvokeCommand(s.dockerHostIP, s.dockerVolumeRemoveCmd)
+	c.Assert(err, IsNil, Commentf(out))
+}
```

Comment:
OK, I will add log statements at end of setup and teardown and that automatically prints current time.
e.g.
```
2017/06/05 17:01:15 SETUP Finished: volumeproperty_test.TestVolumeProperties
2017/06/05 17:01:40 CLEAN-UP Finished: volumeproperty_test.TestVolumeProperties
```